packages/synthetix-main/contracts/modules/core/RewardsManagerModule.sol

```diff
@@ -28,14 +28,11 @@ contract RewardsManagerModule is IRewardsManagerModule {
     // Associated Rewards
     // ---------------------------------------
 
-    function setRewardsDistribution(
+    function registerRewardsDistribution(
         uint128 poolId,
         address collateralType,
         uint index,
-        address distributor,
-        uint amount,
-        uint start,
-        uint duration
+        address distributor
     ) external override {
         if (index > _MAX_REWARD_DISTRIBUTIONS) {
             revert InvalidParameters("index", "too large");
```

Comment:
Further down, where we can only do 'next index', I follow why it's implemented this way, but I wonder if we can just treat them like slots and allow people to insert wherever? Or have it auto-increment? I can think on this more.